satellite/overlay/service.go

```diff
@@ -276,15 +278,23 @@ type Service struct {
 }
 
 // NewService returns a new Service.
-func NewService(log *zap.Logger, db DB, config Config) *Service {
+func NewService(log *zap.Logger, db DB, config Config) (*Service, error) {
+	if config.Node.AsOfSystemTime.Enabled {
+		if config.Node.AsOfSystemTime.DefaultInterval >= 0 {
+			return nil, errs.New("AS OF SYSTEM TIME interval must be a negative number")
+		}
+		if config.Node.AsOfSystemTime.DefaultInterval > -time.Microsecond {
+			return nil, errs.New("AS OF SYSTEM TIME interval cannot be in nanoseconds")
+		}
```

Comment:
Not a necessary change, but more curious your thoughts... would it be better for the AsOfSystemTime struct to have a `isValid()` method to check these things or something like that?